examples/plot_line_testing.py

```diff
@@ -0,0 +1,134 @@
+"""
+Test Line Plots with Matplotcheck
+=================================
+
+These are some examples of using the basic functionality of Matplotcheck
+to test line plots (including regression lines) in Python.
+"""
+
+################################################################################
+# Import Packages
+# ---------------
+# You will start by importing the required packages and plotting a line plot.
+
+import matplotlib.pyplot as plt
+import seaborn as sns
+import numpy as np
+from scipy import stats
+import pandas as pd
+import matplotcheck.base as pt
+
+################################################################################
+# Create Example Data
+# -------------------
+# Before you create a plot, you need to create some data. For plots with
+# regression lines, you will need data in which you are looking for trends
+# over time, such as maximum values from lidar-derived measurements.
+
+# Create dataframe of data points
+col1 = list(np.random.randint(25, size=15))
+col2 = list(np.random.randint(25, size=15))
+data = pd.DataFrame(list(zip(col1, col2)), columns=['Data1', 'Data2'])
+
+# Plot data points, regression line, and one-to-one (1:1) line for reference
+fig, ax = plt.subplots()
+
+# Points and regression line
+sns.regplot('Data1', 'Data2',
+            data=data,
+            color='purple',
+            ax=ax)
+
+# 1:1 line
+ax.plot((0, 1), (0, 1), transform=ax.transAxes, ls='--', c='k')
+
+ax.set(xlabel='Data1',
+       ylabel='Data2',
+       title='Example Data Regression Plot',
+       xlim=(0, 25),
+       ylim=(0, 25))
+
+plt.show()
+
+################################################################################
+# Test Line Plots Using a matplotcheck.PlotTester Object
+# ------------------------------------------------------------
+# Now you can make a ``matplotcheck.PlotTester`` object and test the line plot. 
+# You can test that the line type is a one to one line or a regression line, 
+# and you can test that the line has the correct y intercept and slope.
+
+# Convert matplotlib plot axes object into a matplotcheck PlotTester object
+line_figure_tests = pt.PlotTester(ax)
+
+################################################################################
+# Test Line Types
+# ---------------
+# There are two line types on the plot above: a one-to-one line for
+# reference and a regression line derived from the data points. You can use the method
+# ``assert_lines_of_type()`` to test if a one to one or regression line
+# (or both line types) are present in the plot. (NOTE: Regression and 1:1 lines are the only types
```

Comment:
@nkorinek i'm confused by this. In theory a line is a line. what does the regression line test actually do that is different from testing any other straight line? i'm just confused by the statement - you can test 1:1 and regression. both are lines so both have slopes and intercepts. are they stored differently in the MPL object? let's just chat a bit so i better understand! can MPC test for any line if there is a slope and intercept given?